Automattic/wp-calypso
# Improve/remove pricing page v0

### Changes proposed in this Pull Request

This PR removes code bits that were only used in v0 of the pricing page, which we want to get rid of.

### Implementation notes

The files diff is a bit scary, so let me recap the main changes for clarity.

Removed folders:
- `client/components/jetpack/card/jetpack-product-card/`
- `client/components/jetpack/card/jetpack-bundle-card/`
- `client/components/jetpack/card/jetpack-plan-card/`
- `client/components/jetpack/card/jetpack-free-card/`
- `client/components/jetpack/card/product-card-placeholder/`
- `client/my-sites/plans/jetpack-plans/plans-column/`
- `client/my-sites/plans/jetpack-plans/product-card/`
- `client/my-sites/plans/jetpack-plans/products-column/`

Removed files:
- `client/my-sites/plans/jetpack-plans/details.tsx`
- `client/my-sites/plans/jetpack-plans/selector.tsx`
- `client/my-sites/plans/jetpack-plans/upsell.tsx`
- `client/my-sites/plans/jetpack-plans/use-has-product-upsell.ts`
- `client/my-sites/plans/jetpack-plans/use-is-loading.ts`
- `client/my-sites/plans/jetpack-plans/with-redirect-to-selector.tsx`

Removed test variant:
- `v0 - Offer Reset` in `client/lib/abtest/active-tests.js` and `client/my-sites/plans/jetpack-plans/abtest.ts`

Updated files:
- Removed unused constants in `client/my-sites/plans/jetpack-plans/constants.ts`
- Removed deprecated routes in `client/my-sites/plans/jetpack-plans/index.ts`
- Removed unused middlewares in `client/my-sites/plans/jetpack-plans/controller.tsx`
- Removed unused styles in `client/my-sites/plans/jetpack-plans/style.scss`
- Removed unused types in `client/my-sites/plans/jetpack-plans/types.ts`
- Removed unused util functions in `client/my-sites/plans/jetpack-plans/utils.ts`

Updated routes:
- Redirected to checkout in Google Analytics upsell (`client/my-sites/site-settings/form-analytics.jsx`)
- Redirected to checkout in video hosting upsell (`client/my-sites/site-settings/media-settings-performance.jsx`)
- Redirected to checkout in search engine optimization upsell (`client/my-sites/site-settings/seo-settings/form.jsx`)
- Redirected to checkout in activity log banners (`client/my-sites/activity/activity-log-banner/intro-banner.jsx` and `client/my-sites/activity/activity-log-banner/upgrade-banner.jsx`)

### Testing instructions

- Download the PR
- Run both Calypso and Jetpack cloud concurrently
- Visit the plans page in both environments
- Check that the 3 remaining iterations (v1, v2, and i5) look like production
- Check that you can't see nor select v0
- Test the following upsells:
    - Visit `/marketing/traffic/:site`, click on the GA upsell (see capture), and check that you land in the checkout page with Security Daily in cart
    - In the same page, click on the search engine optimization upsell (see capture), and check that you land in the checkout page with Security Daily in cart
    - Visit `/settings/performance/:site`, click on the video hosting upsell (see capture), and check that you land in the checkout page with Security Daily in cart
    - Visit `/activity-log/:site` with the `dismissible-card-activity-introduction-banner` preference enabled, then disabled (see capture), and check that after clicking the primary button in the banner (see captures), you land in the checkout page with Security Daily in cart
- Make sure tests pass in the CI

### Screenshots

GA upsell
<img width="1064" alt="Screen Shot 2020-11-26 at 12 50 11 PM" src="https://user-images.githubusercontent.com/1620183/100381619-5a11d100-2fe7-11eb-9503-a9ca5b46dbc7.png">

Search engine optimization upsell 
<img width="1067" alt="Screen Shot 2020-11-26 at 12 50 05 PM" src="https://user-images.githubusercontent.com/1620183/100381641-6433cf80-2fe7-11eb-9227-1ac64a806bdb.png">

Video hosting upsell
<img width="760" alt="Screen Shot 2020-11-26 at 12 50 21 PM" src="https://user-images.githubusercontent.com/1620183/100381662-6c8c0a80-2fe7-11eb-8a54-6472168c98c3.png">

Activity log with preference disabled
<img width="870" alt="Screen Shot 2020-11-26 at 1 03 56 PM" src="https://user-images.githubusercontent.com/1620183/100381835-dd332700-2fe7-11eb-914f-e997a85dfede.png">


Activity log with preference enabled
<img width="1088" alt="Screen Shot 2020-11-26 at 12 49 44 PM" src="https://user-images.githubusercontent.com/1620183/100381851-e7edbc00-2fe7-11eb-9ddd-28c2f74c1c2f.png">

Preferences menu
![screenshot](https://user-images.githubusercontent.com/1620183/100381916-19ff1e00-2fe8-11eb-999a-065d3ac0cf9c.png)
